src/modal.ts

```diff
@@ -0,0 +1,462 @@
+
+    @state()
+    _animate_breakpoints = this.DEFAULT_VALUES.animate_breakpoints;
+
+    @state()
+    _guide_value:{ [key:string]:any} = {};
+
+    @state()
+    _open = false;
+
+    @state()
+    _mouse_pos_mod = null;
+
+    @state()
+    _change_mc = false;
+
+    @state()
+    timer_obj:ReturnType<typeof setTimeout>|null = null;
+
+    @query('#__wrapper')
+    _wrapper_el:HTMLElement;
+
+    actualParent:HTMLElement|null;
+
+    private _mouse_up_listener:any;
+    private _mouse_move_listener:any;
+    private _mouse_prev_x:number;
+    private _mouse_prev_y:number;
+    private _wrp_init_bounds:any;
+
+    override attributeChangedCallback(name:string, old_val:string,
+                                      new_val:string) {
+        super.attributeChangedCallback(name, old_val, new_val);
+        if (name == 'open') {
+            requestAnimationFrame(() => {
+             this.open?this.show():this.hide()
+            });
+        }
+        if (name == 'height'){
+            this._height_breakpoints = getResponsiveValues(new_val,
+                                       this.DEFAULT_VALUES.height);
+        }
+        if (name == 'width')
+            this._width_breakpoints = getResponsiveValues(new_val,
+                                      this.DEFAULT_VALUES.width);
+        if(name == 'pos') {
+            this._position_breakpoints = getResponsiveValues(new_val,
+                                         this.DEFAULT_VALUES.pos,
+                (v:string) => {
+                    return position_defs[v] != null;
+                });
+        }
+        if(name == 'animation') {
+            this._animate_breakpoints = getResponsiveValues(new_val,
+                                        this.DEFAULT_VALUES.animation,
+                (v:string) => {
+                    return animate_defs[v] != null;
+            });
+        }
+        if(name == 'guidesel') {
+            let tmp = parseAttrVal(new_val);
+            this._guide_value = {}
+            for (let k of Object.keys(tmp))
+                this._guide_value[tmp[k]] = k;
+        }
+    }
+    constructor() {
+        super();
+        document.addEventListener("keydown", this.handleKey.bind(this));
+    }
+    override connectedCallback() {
+        super.connectedCallback();
+    }
+    override disconnectedCallback() {
+        this.clearTimer();
+    }
+    handleKey(e: KeyboardEvent) {
+        if(!this.noesc && this.open && e.key === 'Escape')
+            if (!active_modals?.length ||
+                active_modals[active_modals.length - 1] == this)
+                this.hide();
+    }
+    clearTimer() {
+        if (this.timer_obj)
+            clearTimeout(this.timer_obj);
+    }
+    containerClicked() {
+        if(this.open && !this.nobgclose)
+            this.hide();
+    }
+    _wrapperClickedEvent(event:any) {
+        event.stopPropagation();
+        if (this.closesel && event.target.matches(this.closesel))
```

Comment:
This event is need only when `closesel` is set. Avoid registering this callback when `closesel` is not set in the first place.